feathr_project/feathr/utils/_env_config_reader.py

```diff
@@ -51,14 +51,20 @@ def get(self, key: str, default: str = None) -> str:
         Returns:
             Feathr client's config value.
         """
-        conf_var = (
-            (self._get_variable_from_env(key) if self.use_env_vars else None) or
-            (self._get_variable_from_file(key) if self.yaml_config else None) or
-            (self._get_variable_from_akv(key) if self.akv_name else None) or
-            default
-        )
+        res1 = (self._get_variable_from_env(key) if self.use_env_vars else None) 
+        res2 = (self._get_variable_from_file(key) if self.yaml_config else None) 
+        res3 = (self._get_variable_from_akv(key) if self.akv_name else None) 
 
-        return conf_var
+        # rewrite the logic below to make sure:
+        # First we have the order (i.e. res1 > res2 > res3 > default)
+        # Also previously we use OR for the result, which will yield a bug where say res1=None, res2=False, res3=None. Using OR will result to None result, although res2 actually have value
+        for res in [res1, res2, res3]:
```

Comment:
This is great. Thank you for identifying the bug.
One minor comment - we can use more meaningful var name here,
for example, res1, res2, res3 -> env_res, file_res, akv_res so that we can easily read how the results are prioritized.